src/Integrators.jl

```diff
@@ -85,6 +91,127 @@ mutable struct ImplicitMidpoint
         return new(Δt, atol)
     end    
 end
+ImplicitMidpoint(; atol) = ImplicitMidpoint(NaN; atol)
+
+function check_timestep_available(integrator)
+    isnan(integrator.Δt) && error("Set integration timestep `Δt`.")
+end
+
+"""
+    suggest_timestep(sys, integrator; tol)
+
+Suggests a timestep for the numerical integration of spin dynamics according to
+a given error tolerance `tol`. The `integrator` should be [`Langevin`](@ref) or
+[`ImplicitMidpoint`](@ref). The suggested ``Δt`` will be inversely proportional
+to the magnitude of the effective field ``|dE/d𝐬|`` arising from the current
+spin configuration in `sys`. The recommended timestep ``Δt`` scales like `√tol`,
+which assumes second-order accuracy of the integrator.
+
+The system `sys` should be initialized to an equilibrium spin configuration for
+the target temperature. Alternatively, a reasonably timestep estimate can be
+obtained from any low-energy spin configuration. For this, one can use
+[`randomize_spins!`](@ref) and then [`minimize_energy!`](@ref).
+
+If `integrator` is of type [`Langevin`](@ref), then the damping magnitude ``λ``
+and target temperature ``kT`` will tighten the timestep bound. If ``λ`` exceeds
+1, it will rescale the suggested timestep by an approximate the factor ``1/λ``.
+If ``kT`` is the largest energy scale, then the suggested timestep will scale
+like ``1/λkT``. Quantification of numerical error for stochastic dynamics is
+subtle. The stochastic Heun integration scheme is weakly convergent of order-1,
+such that errors in the estimates of averaged observables may scale like ``Δt``.
+This implies that the `tol` argument may actually scale like the _square_ of the
+true numerical error, and should be selected with this in mind.
+"""
+function suggest_timestep(sys::System{N}, integrator::Langevin; tol) where N
+    (; Δt, λ, kT) = integrator
+    suggest_timestep_aux(sys; tol, Δt, λ, kT)
+end
+function suggest_timestep(sys::System{N}, integrator::ImplicitMidpoint; tol) where N
+    (; Δt) = integrator
+    suggest_timestep_aux(sys; tol, Δt, λ=0, kT=0)
```

Comment:
Do you want to put a factor of 2 in here? That has been the rule of thumb I've generally used for setting the ImplicitMidpoint time step.